@tlrmchlsmth 
Getting a seemingly unrelated test failure. I saw this in other people's PR as well
```
<head></head>
[2025-05-06T03:45:25Z] =========================== short test summary info ============================
--
  | [2025-05-06T03:45:25Z] FAILED entrypoints/openai/test_audio.py::test_chat_streaming_audio[https://vllm-public-assets.s3.us-west-2.amazonaws.com/multimodal_asset/mary_had_lamb.ogg-fixie-ai/ultravox-v0_5-llama-3_2-1b] - AssertionError: assert 'This audio a...t from a poem' == 'This audio a...a traditional'
  | [2025-05-06T03:45:25Z]
  | [2025-05-06T03:45:25Z]   - This audio appears to be a snippet from a traditional
  | [2025-05-06T03:45:25Z]   ?                                           ^^^^^^^ ^^^
  | [2025-05-06T03:45:25Z]   + This audio appears to be a snippet from a poem
  | [2025-05-06T03:45:25Z]   ?                                           ^ ^^
  | [2025-05-06T03:45:25Z] ====== 1 failed, 420 passed, 2 skipped, 36 warnings in 3645.43s (1:00:45) ======
  | [2025-05-06T03:45:28Z] 🚨 Error: The command exited with status 1
  | [2025-05-06T03:45:28Z] user command error: The plugin docker command hook exited with status 1


```

--- a/vllm/model_executor/layers/mamba/mamba2_metadata.py
+++ b/vllm/model_executor/layers/mamba/mamba2_metadata.py
@@ -71,14 +71,14 @@ def prepare_mamba2_metadata(
 
     # Need flags to indicate if there are initial states
     # currently we really only support the FlashAttention backend
-    # initial states are only relevant for prefills
     has_initial_states = None
     prep_initial_states = False
     if (isinstance(attn_metadata, (FlashAttentionMetadata, XFormersMetadata,
                                    PlaceholderAttentionMetadata))
             and attn_metadata.context_lens_tensor is not None):
+        # keeping flags for both prefill and decode causal_conv1d varlen
         has_initial_states = attn_metadata.context_lens_tensor > 0  # [batch,]
-        # precompute flag to avoid device syncs later in mamba2 forwards
+        # precompute flag to avoid device syncs later in mamba2 layer forwards
         # prep is only needed for mamba2 ssd prefill processing
         prep_initial_states = torch.any(
             has_initial_states[:num_prefills]).item()

--- a/vllm/model_executor/layers/mamba/mamba2_metadata.py
+++ b/vllm/model_executor/layers/mamba/mamba2_metadata.py
@@ -77,7 +77,9 @@ def prepare_mamba2_metadata(
             and attn_metadata.context_lens_tensor is not None):
         has_initial_states = attn_metadata.context_lens_tensor > 0  # [batch,]
         # precompute flag to avoid device syncs later in mamba2 forwards
-        prep_initial_states = torch.any(has_initial_states).item()
+        # prep is only needed for mamba2 ssd prefill processing
+        prep_initial_states = torch.any(
+            has_initial_states[:num_prefills]).item()
 
     # Compute seq_idx, chunk_indices and chunk_offsets for prefill only
     seq_idx = None

--- a/vllm/model_executor/layers/mamba/ops/ssd_combined.py
+++ b/vllm/model_executor/layers/mamba/ops/ssd_combined.py
@@ -40,7 +40,6 @@ def _mamba_chunk_scan_combined_fwd(x,
     _, _, ngroups, dstate = B.shape
     assert nheads % ngroups == 0
     assert B.shape == (batch, seqlen, ngroups, dstate)
-    assert x.shape == (batch, seqlen, nheads, headdim)
     assert dt.shape == (batch, seqlen, nheads)
     assert A.shape == (nheads, )
     assert C.shape == B.shape

